src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -0,0 +1,231 @@
+              optional = true)
+    public boolean useSupplementaryAlignments = false;
+
+    @Argument(doc = "If set non-zero value, only include reads passing certain mapping quality threshold. " +
+                    "If set to zero, reads with zero mapping quality will be included in calculating metrics.",
+              shortName = "Q",
+              fullName = "MQPassingThreshold",
+              optional = true)
+    public int MQPassingThreshold = 0;
+
+//    TODO
+//    @Argument(doc = "If set to non-zero value, only include reads with mate passing certain mapping quality threshold.",
+//              shortName = "Qm",
+//              fullName = "mateMQPassingThreshold",
+//              optional = true)
+//    public int mateMQPassingThreshold = 0;
+
+//    @Argument(doc="The level(s) at which to accumulate metrics. Possible values are {SAMPLE, LIBRARY, READ GROUP}. " +
+//                  "Default: All reads.",
+//              shortName="LEVEL",
+//              fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+//              optional = false)
+//    public Set<MetricAccumulationLevel> METRIC_ACCUMULATION_LEVEL = EnumSet.of(MetricAccumulationLevel.ALL_READS);
+
+    /**
+     * Which end of a read pair to use for collecting insert size metrics.
+     * TODO: For truncated genomic regions of putative SV breakpoints, not all reads have both ends land in the region,
+     *       so third case is possible: will use either end when only one end is available in the region specified,
+     *       and only first end if both are available
+     */
+    public enum whichEndToUse{
+        first(1), second(2);//, either(0);
+        private final int value;
+        whichEndToUse(int value){
+            this.value = value;
+        }
+        public int getValue(){
+            return value;
+        }
+    }
+
+    @Argument(doc = "Which end of pairs to use for collecting information. Possible values:{0,1,2}, where 1 and 2 " +
+                    "stands for first or second end respectively. Option 0 picks up information from 1st end " +
+                    "when both ends are available, and pick either one when only one end is available. " +
+                    "(Remember, in SV analysis, a truncated region may be investigated, so this is possible.)" +
+                    " Unused if only paired reads are used (properlyPairedOnly==true).",
+              shortName = "F",
+              fullName = "whichEndOfPairToUse",
+              optional = true)
+    public whichEndToUse useEnd = whichEndToUse.first;
+
+    // path to Picard R script for producing histograms in PDF files.
+    private static final String R_SCRIPT = "insertSizeHistogram.R";
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<GATKRead> filteredReads = getReads();
+
+        final long readCount = filteredReads.count();
+
+        if(null == filteredReads || 0 == readCount){
```

Comment:
well, this is impossible here `filteredReads == null` because the line above would have blown up.